### ISSUE TITLE
malloc_hook.multi_threads failure with valgrind

### ISSUE THREAD

**Issue by yosefe** [openucx/ucx#1280]
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2786/label=hpc-test-node/consoleFull
```
15:42:03 [ RUN      ] malloc_hook.multi_threads
15:42:04 [     INFO ] Reduced heap free space to minimum: 1485264 after allocating 2850000 bytes
15:42:04 ==25301== Warning: client syscall shmdt tried to modify addresses 0xffffffffffffffff-0xffffffffffffffff
15:42:30 [     INFO ] thread 5/8: small mapped: 512000, total mapped: 1014804480, total unmapped: 1014501376
15:42:30 [     INFO ] thread 7/8: small mapped: 503808, total mapped: 1012781056, total unmapped: 1014534144
15:42:30 [     INFO ] thread 2/8: small mapped: 503808, total mapped: 1013284864, total unmapped: 1014534144
15:42:30 [     INFO ] thread 4/8: small mapped: 503808, total mapped: 1014292480, total unmapped: 1014534144
15:42:30 [     INFO ] thread 3/8: small mapped: 503808, total mapped: 1013788672, total unmapped: 1014534144
15:42:30 [     INFO ] thread 0/8: small mapped: 499712, total mapped: 1015803904, total unmapped: 1014542336
15:42:30 [     INFO ] thread 6/8: small mapped: 499712, total mapped: 1015304192, total unmapped: 1014550528
15:42:30 /scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node/contrib/../test/gtest/ucm/malloc_hook.cc:269: Failure
15:42:30 Value of: is_ptr_in_range(ptr, shm_seg_size, m_unmap_ranges)
15:42:30   Actual: false
15:42:30 Expected: true
15:42:30 [     INFO ] thread 1/8: small mapped: 499712, total mapped: 1016303616, total unmapped: 1014550528
15:42:30 
[  FAILED  ] malloc_hook.multi_threads, where TypeParam =  and GetParam() =  (26870 ms)
15:42:30 [ RUN      ] malloc_hook.single_thread
```

**Comment by yosefe** [comment]
happens because pr #1254 allowed concurrency between events

**Comment by yosefe** [comment]
happened with http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2971/label=hpc-test-node2,worker=1/consoleFull

**Comment by yosefe** [comment]
also w/o valgrind:
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3082/label=hpc-test-node,worker=1/consoleFull
```
23:42:42 [     INFO ] Using random seed of 4690
23:42:42 Note: This is test shard 2 of 4.
23:42:42 Note: Randomizing tests' orders with a seed of 11654 .
23:42:42 [==========] Running 1465 tests from 390 test cases.
...
23:52:58 [----------] 1 test from malloc_hook, where TypeParam = 
23:52:58 [ RUN      ] malloc_hook.multi_threads
23:52:58 [     INFO ] Reduced heap free space to minimum: 5015312 after allocating 10000 bytes
23:52:58 /scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node/worker/1/contrib/../test/gtest/ucm/malloc_hook.cc:276: Failure
23:52:58 Value of: is_ptr_in_range(ptr, shm_seg_size, m_unmap_ranges)
23:52:58   Actual: false
23:52:58 Expected: true
23:52:58 [     INFO ] thread 4/8: small mapped: 23576576, total mapped: 873639936, total unmapped: 758292480
23:52:58 [     INFO ] thread 3/8: small mapped: 35966976, total mapped: 936529920, total unmapped: 842223616
23:52:58 [     INFO ] thread 6/8: small mapped: 16084992, total mapped: 999514112, total unmapped: 926806016
23:52:58 [     INFO ] thread 5/8: small mapped: 56549376, total mapped: 994197504, total unmapped: 926814208
23:52:58 [     INFO ] thread 0/8: small mapped: 26202112, total mapped: 980836352, total unmapped: 926814208
23:52:58 [     INFO ] thread 2/8: small mapped: 76627968, total mapped: 999514112, total unmapped: 926814208
23:52:58 [     INFO ] thread 7/8: small mapped: 72691712, total mapped: 1041469440, total unmapped: 973647872
23:52:58 [     INFO ] thread 1/8: small mapped: 33501184, total mapped: 1007865856, total unmapped: 979234816
23:52:58 
[  FAILED  ] malloc_hook.multi_threads, where TypeParam =  and GetParam() =  (246 ms)
23:52:58 [----------] 1 test from malloc_hook (246 ms total)
```

**Comment by yosefe** [comment]
fixed by #1482
